libpromises/eval_context.c

```diff
@@ -1865,10 +1871,7 @@ VariableTableIterator *EvalContextVariableTableFromRefIteratorNew(const EvalCont
 
 const void *EvalContextVariableControlCommonGet(const EvalContext *ctx, CommonControl lval)
 {
-    if (lval >= COMMON_CONTROL_MAX)
-    {
-        return NULL;
-    }
+    assert(lval >= 0 && lval < COMMON_CONTROL_MAX);
```

Comment:
``assert`` is debug-only. Maybe keep the ``return NULL`` for release builds, or else make it ``ProgrammingError`` instead?